src/python/pants/backend/java/test/junit.py

```diff
@@ -32,40 +29,40 @@ class JavaTestFieldSet(TestFieldSet):
     sources: JavaTestSourceField
 
 
+class JunitTool(JvmToolBase):
+    options_scope = "junit-tool"
+    help = "JUnit tool configuration"
+
+    default_version = "5.7.2"
+    default_artifacts = [
+        "org.junit.platform:junit-platform-console:1.7.2",
```

Comment:
Unclear to me how best to handle multiple different versions in the artifacts vis-a-vis the `--version` option.

Reply:
Do you mean collisions between the tool and the user's classpath? Eventually we'll likely need to implement shading to handle that: #13259.

Reply:
No, just the artifacts having different versions. Note that the "main" version number is 5.7.2 but the `junit-platform-console` artifact has version 1.7.2. Maybe `--version` arguments should be defined by the individual tools based on their needs?

Reply:
Oh, I see what you are saying. Hm. It may be better not to include the version yet then...? It's an easy feature to add later, and it is pretty error prone here (although I do think that this case is a bit exceptional).